src/pxfw.cpp

```diff
@@ -0,0 +1,654 @@
+
+	dev->cmd[0] = 0xF1;
+	dev->cmd[1] = 0x01;
+	dev->cmd[7] = idx;
+	dev->cmd[8] = (sz >> 8) & 0xFF;
+	dev->cmd[9] = sz & 0xFF;
+	if ((dev->err=dev->cmd.transport(READ,buf,sz) ))
+		{ sperror ("read EEPROM",dev->err); 	return (0); }
+
+	printf("EEPROM block #%d:\n",idx);
+	for(i=0;i<(sz/0x10);i++) {
+	    printf("| %X0 | ", i);
+	    for(j=0;j<0x10;j++) printf("%02X ",buf[i*0x10+j]);
+	    printf("|");
+	    for(j=0;j<0x10;j++) {
+		if (buf[i*0x10+j] > 0x20) printf("%c",buf[i*0x10+j]);
+		else printf(" ");
+	    }
+	    printf("|\n");
+	};
+	return 1;
+}
+*/
+
+int fwblk_send(drive_info* dev, int offs, int blksz, bool last) {
+	dev->cmd[0] = 0x3B;
+	dev->cmd[1] = last ? 0x05:0x04;
+	dev->cmd[2] = 0x00;
+	dev->cmd[3] = (offs >> 16) & 0xFF;
+	dev->cmd[4] = (offs >> 8) & 0xFF;
+	dev->cmd[5] = offs & 0xFF;
+	dev->cmd[6] = (blksz >> 16) & 0xFF;
+	dev->cmd[7] = (blksz >> 8) & 0xFF;
+	dev->cmd[8] = blksz & 0xFF;
+	dev->cmd[9] = 0x00;
+	dev->cmd[10]= 0x00;
+	dev->cmd[11]= 0x00;
+
+	if ((dev->err=dev->cmd.transport(WRITE,dev->rd_buf,blksz) )){
+		sperror ("SEND_FWBLK",dev->err);
+		if (!last) return 1;
+	}
+	return 0;
+}
+
+
+int fwblk_read(drive_info* dev, int offs, int blksz, unsigned char flag) {
+        printf("data read...\n");
+	return 1;
+}
+
+/*
+	dev->cmd[0] = 0x3B;
+//	cmd[1] = last ? 0x05:0x04;
+	dev->cmd[1] = flag;
+
+	dev->cmd[2] = 0x00;
+	dev->cmd[3] = (offs >> 16) & 0xFF;
+	dev->cmd[4] = (offs >> 8) & 0xFF;
+	dev->cmd[5] = offs & 0xFF;
+	dev->cmd[6] = (blksz >> 16) & 0xFF;
+	dev->cmd[7] = (blksz >> 8) & 0xFF;
+	dev->cmd[8] = blksz & 0xFF;
+	dev->cmd[9] = 0x00;
+	dev->cmd[10]= 0x00;
+	dev->cmd[11]= 0x00;
+
+	if ((dev->err=dev->cmd.transport(READ,dev->rd_buf,blksz) )){
+		sperror ("READ_FWBLK",dev->err);
+		return 1;
+	}
+	return 0;
+}
+*/
+
+void csum_init(unsigned short int *csum, int FW_dev_ID){
+    *csum = 0;
+    if (FW_dev_ID <= PLEXTOR_PREMIUM) *csum = 0x8000;
+}
+
+void csum_update(unsigned short int *csum, unsigned char* buf, int len) {
+	int		i;
+	unsigned int	ad;
+	for (i=0; i<len; i++)
+		{ ad = buf[i]; *csum += ad; }
+}
+
+void usage(char* bin) {
+	fprintf (stderr,"\nusage: %s [-d device] [options]\n",bin);
+	printf("\t-h\t\t show help\n");
+	printf("\t-v\t\t debug\n");
+	printf("\t-l\t\t scan IDE/SCSI bus\n");
+	printf("\t-u\t\t write FW\n");
+	printf("\t-if [file]\t read FW from this file\n");
+//	printf("\t-of [file]\t save FW to this file\n");
+	printf("\t-f\t\t force flashing, even if device not recognized or checksum error\n");
+	printf("\t-e\t\t read EEPROM\n");
+//	printf("\t-ie [file]\t load EEPROM from this file\n");
+	printf("\t-oe [file]\t save EEPROM to this file\n");
+	printf("\t-r\t\t reset device\n");
+	printf("\t-t\t\t test opcodes supported by devive (EXPERIMENTAL),\n");
+	printf("\t\t\t use it only if you know what are you doing!\n");
+}
+
+int main(int argc,char *argv[]) {
+	drive_info  *dev=0;
+	char        *devname=0;
+	char	    *fwfname=0;
+	char	    *eefname=0;
+	FILE	    *fwfile;
+	FILE	    *eefile;
```

Comment:
Codacy found an issue: [The scope of the variable 'eefile' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)